node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/yum/Yum.java

```diff
@@ -0,0 +1,83 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.hosted.node.admin.task.util.yum;
+
+import com.yahoo.vespa.hosted.node.admin.component.TaskContext;
+import com.yahoo.vespa.hosted.node.admin.task.util.process.ChildProcess;
+import com.yahoo.vespa.hosted.node.admin.task.util.process.Command;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.logging.Logger;
+
+public class Yum {
+    private static Logger logger = Logger.getLogger(Yum.class.getName());
+
+    private final TaskContext taskContext;
+    private final Supplier<Command> commandSupplier;
+    private List<String> packages = new ArrayList<>();
+
+    public Yum(TaskContext taskContext) {
+        this.taskContext = taskContext;
+        this.commandSupplier = () -> new Command(taskContext);
+    }
+
+    /**
+     * @param packages A list of packages, each package being of the form name-1.2.3-1.el7.noarch
+     */
+    public Install install(String... packages) {
+        return new Install(taskContext, Arrays.asList(packages));
+    }
+
+    public class Install {
+        private final TaskContext taskContext;
+        private final List<String> packages;
+        private Optional<String> enabledRepo = Optional.empty();
+
+        public Install(TaskContext taskContext, List<String> packages) {
+            this.taskContext = taskContext;
+            this.packages = packages;
+
+            if (packages.isEmpty()) {
+                throw new IllegalArgumentException("No packages specified");
+            }
+        }
+
+        public Install enableRepo(String repo) {
+            enabledRepo = Optional.of(repo);
+            return this;
+        }
+
+        public boolean converge() {
+            if (packages.stream().allMatch(Yum.this::isInstalled)) {
```

Comment:
Would be nice if we could execute the command once to get a list of all installed packages and then just compare two `Set`s.

Reply:
Yes, but that would require parsing the output. Now, it's just an exit code.